### PR TITLE
Adding setup with docker

### PR DESCRIPTION
**Pull request by laurauzcategui**
Hi @alexeygrigorev , 

 I created a small guide for anyone who feels comfortable using Docker or might want to try it for setting up the environment. 

 Since I saw a couple of questions today related to environment setup, I thought of sharing what I usually use when working on projects or courses, then it can be re-usable. 

Hoping is helpful :) 

Changelog: 

- Updated readme with link to guide to create docker container
- Added new guide to build docker container and run it
- Added Dockerfile and environment.yml

### CODE REVIEW

**Review by aurcode — APPROVED** [review]
It's good idea

**Comment by alexeygrigorev** [comment]
That looks really great, thank you! Maybe you could put it to the course folder? 

E.g. to `course-zoomcamp/01-intro`, create a folder there "environment" and reference it from `06-environment.md`?

**Comment by laurauzcategui** [comment]
> That looks really great, thank you! Maybe you could put it to the course folder?
> 
> E.g. to `course-zoomcamp/01-intro`, create a folder there "environment" and reference it from `06-environment.md`?

Cool, will update the PR :)

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+RUN $HOME/anaconda3/bin/conda env create -f environment.yml \
+ && $HOME/anaconda3/bin/conda clean -ya
+ENV CONDA_DEFAULT_ENV=py38
+ENV CONDA_PREFIX=$HOME/anaconda3/envs/$CONDA_DEFAULT_ENV
+ENV PATH=$CONDA_PREFIX/bin:$PATH 
+RUN $HOME/anaconda3/bin/conda clean -ya
+RUN echo "source activate $(head -1 environment.yml | cut -d' ' -f2)" > ~/.bashrc
+ENV PATH /opt/conda/envs/$(head -1 environment.yml | cut -d' ' -f2)/bin:$PATH
+
+# Install Tensorflow and XgBoost
```
Comment (alexeygrigorev) [inline Dockerfile]: where's xgboost? =) you can also install them with one pip command, e.g. `pip install xgboost tensorflow` 

(why did you include grpcio?)

File: appendix-a-setup-docker.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+This will open jupyter in your browser. 
+
+## How to add more libraries or update a version? 
+
+Follow the next steps.
+
+1. Go to the [environments.yml](./environment.yml)
+
+You should see a list as followS:
```
Comment (alexeygrigorev) [inline appendix-a-setup-docker.md]: capital S at the end

File: environment.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+dependencies:
+  - python=3.8
+  - jupyter
+  - numpy
+  - pandas
+  - matplotlib
+  - scikit-learn
+  - seaborn
+  - cudatoolkit
+  - xgboost
```
Comment (alexeygrigorev) [inline environment.yml]: oh I see why you didn't do `pip install xgboost`

Comment (laurauzcategui) [reply environment.yml]: Yes, Added on conda for simplicity and to have it all in one place.

File: Dockerfile (hunk 1 above)
Comment (laurauzcategui) [reply Dockerfile]: Got a dependency problem when trying to install Tensorflow  or Xgboost and figured it out by looking at someone having the same issue. If I remember correctly, there is an issue with version of grpcio.

File: appendix-a-setup-docker.md (hunk 2 above)
Comment (laurauzcategui) [reply appendix-a-setup-docker.md]: Thanks

File: Dockerfile (hunk 1 above)
Comment (laurauzcategui) [reply Dockerfile]: I reproduced again and found out it actually need python-dev and build-essential and with those there is no need to fix grpcio version. 

Thanks for the catch @alexeygrigorev

**Comment by alexeygrigorev** [comment]
Thank you! Is it ready?

**Comment by laurauzcategui** [comment]
> Thank you! Is it ready?

Yes! All good. 

Ship it ☺️

**Comment by alexeygrigorev** [comment]
Thank you for your contribution!

**Pull request merged**